### PR TITLE
TE-1576 Hide navigation items if they exceed 900px on desktop

### PR DESCRIPTION
**Pull request by red-runner**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1576)

### What **one** thing does this PR do?
Hides the `Header` if  the navigation items exceed 900px on computer and widescreen screens; and, the `Header` will display a mobile-style burger menu in all viewports.

### Any other notes
large_navigation_items

### CODE REVIEW

File: src/components/collections/Header/utils/getStandardMenuMarkup.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+export const getStandardMenuMarkup = (
+  {
+    /* eslint-disable react/prop-types */
+    activeNavigationItemIndex,
+    navigationItems,
+    primaryCTA,
+    /* eslint-enable react/prop-types */
+  },
+  isStandardMenuShowingOnComputer
```
Comment (marcdavi-es) [inline src/components/collections/Header/utils/getStandardMenuMarkup.js]: Do we need this? 

I think it's always true. https://github.com/lodgify/lodgify-ui/pull/473/files#diff-ed07c99ab29b1b82da1000fb02576851R40

If that's right, we don't need to change any logic in this file, just the name.

File: src/components/collections/Header/Readme.md [hunk 2]
```diff
@@ -153,3 +153,52 @@ const logoSrc = require('./mock-data/livingstoneLogo.png');
   primaryCTA={{ onClick: console.log, text: 'Book now'}}
 />
 ```
+
+#### Large Navigation items
```
Comment (marcdavi-es) [inline src/components/collections/Header/Readme.md]: Maybe `Wide navigation items` is better?

File: src/components/collections/Header/component.js [hunk 3]
```diff
@@ -27,8 +28,18 @@ export const Component = props => (
...
         props.logoSrcSet
       )}
-      {getMobileMenuMarkup(props)}
-      {getDesktopMenuMarkup(props)}
+      {areNavigationItemsExceedingHeaderMaxWidth(
+        HEADER_MAX_WIDTH,
+        props.navigationItems,
+        props.primaryCTA
+      ) ? (
+        <Fragment>{getHiddenMenuMarkup(props, true)}</Fragment>
```
Comment (marcdavi-es) [inline src/components/collections/Header/component.js]: I don't think we need the `Fragment` wrapper here?

File: src/components/collections/Header/component.js [hunk 4]
```diff
@@ -27,8 +28,18 @@ export const Component = props => (
...
+      ) : (
+        <Fragment>
+          {getHiddenMenuMarkup(props, false)}
+          {getStandardMenuMarkup(props, true)}
```
Comment (marcdavi-es) [inline src/components/collections/Header/component.js]: See my comment on `getStandardMenuMarkup`. If you agree, we can remove the second argument here.

File: src/components/collections/Header/mock-data/navigationItems.js [hunk 5]
```diff
@@ -21,4 +21,37 @@ export const navigationItems = [
       },
     ],
   },
+  { text: 'Contact Us', href: '/' },
+  { text: 'What to do', href: '/' },
+];
+
+export const LargeNavigationItems = [
```
Comment (marcdavi-es) [inline src/components/collections/Header/mock-data/navigationItems.js]:
```suggestion
export const largeNavigationItems = [
```

File: src/components/collections/Header/utils/getStandardMenuMarkup.js (hunk 1 above)
Comment (red-runner) [reply src/components/collections/Header/utils/getStandardMenuMarkup.js]: No we don't. It's always true 👍

File: src/components/collections/Header/utils/areNavigationItemsExceedingHeaderMaxWidth.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import { getPrimaryCTAWidth } from './getPrimaryCTAWidth';
+
+/**
+ * @param  {number} headerMaxWidth
+ * @param  {Array}  navigationItems
+ * @param  {Object} primaryCTA
+ * @return {boolean}
+ */
+export const areNavigationItemsExceedingHeaderMaxWidth = (
+  headerMaxWidth,
```
[the remark this answers is not in the document]
Comment (red-runner) [reply src/components/collections/Header/utils/areNavigationItemsExceedingHeaderMaxWidth.js]: @Jonnygwi I thought that there was no harm in making this constant explicit in the component. This way it's very easy to know which factors we are taking into account inside `areNavigationItemsExceedingHeaderMaxWidth`.  What do you think?

[the remark this answers is not in the document]
Comment (Jonnygwi) [reply src/components/collections/Header/utils/areNavigationItemsExceedingHeaderMaxWidth.js]: Yeah I hear what you are saying @red-runner . The name of the function also has `HeaderMaxWidth` within it so it's clear from that what the function is depending on.

**Pull request merged**